### ISSUE TITLE
mod_security bug or misbehaving

### ISSUE THREAD

**Issue by gilperon** [SpiderLabs/ModSecurity#988]
I have this conf:

    SecRuleEngine On
    SecRule REQUEST_URI "kkkk" "phase:1,t:none,noauditlog,log,deny,status:429,msg:'CCCCC',id:234322"

It works fine, I mean, when the user hits "http://example.com?kkkkkk" I can see that mod_security is logging on error_log the string "CCCCC".

However when the user hits "http://example.com/kkkkkk" apache returns 404 error cause this directory does not exist BUT mod_security does not log it - actually mod_security looks to do nothing when the returned status of apache is 404. How can I fix this bug?

**Comment by csanders-git** [comment]
I can't actually tell you why this isn't working. From the information your provided it looks like it should catch on both. Perhaps try making the rule slightly more readable by adding @contains instead of the implied @rx, but even so this rule should work. Can you provide debug logs if this is still an issue?

**Comment by gilperon** [comment]
You are right, that's why I closed this topic 5 days ago so it should not consume your time! Thank you for your great work!

I would just like to have an answer that keeps bogging me for a long time: I have a mod_security script that initializes a variable :xx=0 and I also increment it everytime a user hits my server. Everything works fine but sometimes the xx variable does not get set to 0. I came to know that the =+ operation is atomic but setting a variable is not. I am working on a really busy server and I think this may be the problem. So how do I make setting a variable to a value (like 0) to be atomic?
